classes/privacy/helper.php

```diff
@@ -0,0 +1,88 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+/**
+ * Privacy Subsystem implementation for mod_moodleoverflow.
+ *
+ * @package    mod_moodleoverflow
+ * @copyright  2018 Tamara Gunkel
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+namespace mod_moodleoverflow\privacy;
+use \core_privacy\local\request\approved_contextlist;
+use \core_privacy\local\request\writer;
+use \core_privacy\local\metadata\item_collection;
+defined('MOODLE_INTERNAL') || die();
+/**
+ * Subcontext helper trait.
+ *
+ * @copyright  2017 Tamara Gunkel
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+trait helper {
+    /**
+     * Get the discussion part of the subcontext.
+     *
+     * @param   \stdClass   $discussion
+     * @return  array
+     */
+    protected static function get_discussion_area(\stdClass $discussion) : Array {
```

Comment:
We support Moodle 3.3, so support for PHP 5.6 is required! Return type declarations are not supported by PHP 5.6.
Apart from that, `array` is supposed to be lowercase: http://php.net/manual/en/functions.arguments.php#functions.arguments.type-declaration.